modules/solid_mechanics/src/userobjects/CrackFrontDefinition.C

```diff
@@ -129,11 +146,12 @@ CrackFrontDefinition::initialize()
 void
 CrackFrontDefinition::finalize()
 {
-}
-
-void
-CrackFrontDefinition::threadJoin(const UserObject & /*uo*/)
-{
+  if (_t_stress)
+  {
+    unsigned int num_crack_front_nodes = _ordered_crack_front_nodes.size();
+    for (unsigned int i=0; i<num_crack_front_nodes; ++i)
+      gatherMax(_strain_along_front[i]);
```

Comment:
This is a rather slow operation. You are doing `num_crack_front_nodes` individual parallel communications. You hopefully can do them all at once.
e.g.
`_communicator.max(_strain_along_front)`
If that isn't it, check `libmesh/include/parallel/parallel.h` for available aggregation operations.